### PR TITLE
Localize URL: Don't rewrite wordpress.com URLs that point to Calypso

### PR DESCRIPTION
**Pull request by akirk**
#### Changes proposed in this Pull Request

* URLs like https://wordpress.com/checkout/offer-quickstart-session/ that go through `localizeUrl()` should not get a locale inserted since they belong to Calypso.

#### Testing instructions

* Unit tests: `cd packages/i18n-utils; yarn test`

See p1615295603002600-slack-i18n

### CODE REVIEW

File: packages/i18n-utils/src/localize-url.tsx [hunk 1]
```diff
@@ -98,7 +98,17 @@ const urlLocalizationMapping: UrlLocalizationMapping = {
 		url.pathname = url.pathname.replace( /\/help\//, '/support/' );
 		return prefixLocalizedUrlPath( supportSiteLocales )( url, localeSlug );
 	},
-	'wordpress.com': prefixLocalizedUrlPath( magnificentNonEnLocales ),
+	'wordpress.com': ( url: URL, localeSlug: Locale ) => {
+		// Don't rewrite checkout and me URLs.
+		if ( /^\/(checkout|me)(\/|$)/.test( url.pathname ) ) {
+			return url;
+		}
+		// Don't rewrite Calypso URLs that have the URL at the end.
```
Comment (yuliyan) [inline packages/i18n-utils/src/localize-url.tsx:106]: Is it supposed to match only the pathnames that have the url as last parameter, because the regex is currently matching when it has url no matter if it is the last parameter or not?

**Review by yuliyan — COMMENTED** [review]
Thanks for fixing it! I got a bit confused from the comment for the case for pathnames with URLs - could you confirm if it's supposed to work like that?

File: packages/i18n-utils/src/localize-url.tsx (hunk 1 above)
Comment (akirk) [reply packages/i18n-utils/src/localize-url.tsx:106]: Not sure I 100% understand the question. This is to match Calypso style URLs that look like mentioned in the unit tests: https://wordpress.com/home/test.wordpress.com, https://wordpress.com/domains/test.wordpress.com etc. One edge case that is currently not rewritten and might need a change would be a URL like https://wordpress.com/test.wordpress.com, do you mean that?

Comment (akirk) [reply packages/i18n-utils/src/localize-url.tsx:106]: Ah, now I understand the mistake in my Regex, I'll update it.

Comment (akirk) [reply packages/i18n-utils/src/localize-url.tsx:106]: Updated. Thanks for catching this!

**Review by yuliyan — APPROVED** [review]
LGTM 🚀
